src/DataWrangling/ECCO/ECCO_metadata.jl

```diff
@@ -243,13 +243,23 @@ function download_dataset(metadata::ECCOMetadata; url = urls(metadata))
         # Write down the username and password in a .netrc file
         downloader = netrc_downloader(username, password, "ecco.jpl.nasa.gov", tmp)
         ntasks = Threads.nthreads()
-        
+       
+        # Check if all file paths exist
+        all_files_exist = all(isfile(metadata_path(metadatum)) for metadatum in metadata)
+
+        # If all files exist, print the message once
+        if all_files_exist
+            println("Note: ECCO $(metadata.name) data is in $(metadata.dir).")
+        else
+            println("Downloading ECCO data: $(metadata.name)")
+        end
```

Comment:
I like this message, but it seems like there is an assumption here. Should we put the downloading into the second branch so the message is guaranteed to be correct? In other words:

```julia
if not_downloading
    # say we aren't
else
    # say we are downloading
    # do the downloading
end
```

Reply:
I'm not sure what assumption you mean? I am using exactly the same if statement as the statement which determines if downloading should occur. The reason I don't print these messages in the if loop where downloading happens is because the message then gets repeated by the numbers of years x months that are downloaded/not downloaded, i.e., you will always see 24 downloaded + not downloaded messages for two years, which feels a bit too verbose. 

Instead, the code checks if _all_ files are present in the specified directory, then, the code says either that files were downloaded or they were not downloaded.

Reply:
> I'm not sure what assumption you mean? I am using exactly the same if statement as the statement which determines if downloading should occur.

Sorry, to clarify, it is only an assumption if the two separate `if` statements continue to use the same criteria in the future. Since you haven't guaranteed that explicitly, future developments can cause this statement to become invalid. You could add a test about printing, but its hard to do that. So better to ensure that future developments don't break it through a slight change in the code.

Reply:
I have added a modification that should account for this - only one conditional is used